backend/app/core/authz/authorization.py

```diff
@@ -52,13 +51,13 @@ def enforce(
         resolver: type[SubjectResolver],
         *,
         object_id: str = "id",
-    ) -> Callable[[Request, User, Session], None]:
-        def inner(
+    ) -> Callable[[Request, User, Session], Awaitable[None]]:
+        async def inner(
```

Comment:
why do you make it async? We used to have an async resolver but switched to sync